src/App.tsx

```diff
@@ -1,20 +1,58 @@
-import React from 'react';
+import React, { ChangeEvent, useCallback, useEffect, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedName, setSelectedName] =
+    useState<string>('No selected person');
+  const [personList, setPersonList] = useState(peopleFromServer);
+  const [inputName, setInputName] = useState('');
+  const [appliedInput, setAppliedInput] = useState('');
+  const [isInputFocus, setIsInputFocus] = useState(true);
+
+  const applyInput = useCallback((value: string) => {
+    debounce(() => setAppliedInput(value), 300)();
```

Comment:
Delay should be customizable through a prop [Reference](https://github.com/mate-academy/react_autocomplete#:~:text=suggestions%20should%20appear,value%20is%20300ms\)%3B)